tests/test_filehashstore_interface.py

```diff
@@ -529,16 +658,17 @@ def test_store_sysmeta_thread_lock(store):
 def test_retrieve_object(pids, store):
     """Test retrieve_object returns correct object data."""
     test_dir = "tests/testdata/"
+    format_id = "http://ns.dataone.org/service/types/v2.0"
     for pid in pids.keys():
         path = test_dir + pid.replace("/", "_")
         filename = pid.replace("/", "_") + ".xml"
         syspath = Path(test_dir) / filename
-        hash_address = store.store_object(pid, path)
-        store.store_sysmeta(pid, syspath)
+        object_metadata = store.store_object(pid, path)
+        store.store_metadata(pid, syspath, format_id)
         obj_stream = store.retrieve_object(pid)
         sha256_hex = store.computehash(obj_stream)
```

Comment:
As discussed in slack, this line causes a test failure, when running tests from command line (output is: `FAILED tests/test_filehashstore_interface.py::test_retrieve_object - ValueError: unsupported hash type SHA-256`). 

Changing this line to:
```
sha256_hex = store.computehash(obj_stream, "sha256")
```
as you suggested, causes the test to pass

Reply:
Thank you for catching this!

`computehash(stream, algorithm)` when called without specifying an algorithm parameter, defaults to None, in which case it will use `self.algorithm` in FileHashStore. That was set to the unmapped value based on the yaml file when initializing HashStore. To resolve this issue, I have set FileHashStore's `algorithm` attribute during the same call as when we set the default algorithms to return in the hex digest map.